src/makeRobot.js

```diff
@@ -38,7 +38,120 @@
  * @return {object}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = Object.create({
+    name,
+    wheels,
+    version,
+
+    get info() {
+      const n = this.name;
+      const v = this.version;
+      const w = this.wheels;
+
+      return `name: ${n}, chip version: ${v}, wheels: ${w}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    moves: [],
+
+    get coords() {
+      let x = 0;
+      let y = 0;
+
+      if (this.moves.length > 0) {
+        this.moves.forEach(move => {
+          switch (move.action) {
+            case 'f':
+              y += move.payload;
+              break;
+            case 'b':
+              y -= move.payload;
+              break;
+            case 'r':
+              x += move.payload;
+              break;
+            case 'l':
+              x -= move.payload;
+              break;
+            case 'e':
+              x = move.payload.x;
+              y = move.payload.y;
+              break;
+          };
+        });
+
+        return {
+          x,
+          y,
+        };
+      } else {
+        return {
+          x,
+          y,
+        };
+      }
+    },
```

Comment:
coords можно сделать объектом.
```suggestion
    coords: {
      x: 0,
      y: 0,
    },
```
И тогда функции движения и эвакуации можно переписать.
```
        goForward(steps = 1) {
      if (steps > 0) {
        this.coords.y += steps;
      };

      return this;
    },
```

Reply:
Да, так можно было сделать, мне просто захотелось по памяти что-то похожее на редакс логику воссоздать. В рамках отдельно взятого  объекта получилось конечно не очень, но зато не скучно провел часок-другой

Reply:
Код должен быть понятным и легко читаемым